lib/sinon/spy-entire-object.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+var getPropertyDescriptor = require("./util/core/get-property-descriptor");
+var walk = require("./util/core/walk");
+
+function spyEntireObject(spy, object) {
+    walk(object || {}, function(prop, propOwner) {
+        // we don't want to spy things like toString(), valueOf(), etc. so we only spy if the object
```

Comment:
Is there any particular reason for us no to spy on those? IMO these should also be spied on; otherwise it feels like we're making decisions on behalf of our users. I agree most users wouldn't need these spied on, but that's up to them.

I'd go for the least astonishment principle here and spy on every single function.

People can also be attaching methods to `Object.prototype` — which is weird, but valid nonetheless.